packages/data-stores/src/i18n/resolvers.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * External dependencies
+ */
+import { apiFetch } from '@wordpress/data-controls';
+import type { APIFetchOptions } from '@wordpress/api-fetch';
+
+import { setLocalizedLanguageNames } from './actions';
+
+export function* getLocalizedLanguageNames( locale: string ) {
+	const localizedLanguageNames = yield apiFetch( {
+		// Forcefully add `_locale` here because the `data` parameter will re-write it as
```

Comment:
I wasn't sure why it rewrites it to just `locale` (no underscore). Could you elaborate?

Furthermore, could we overwrite the default `userLocaleMiddleware` middleware with our own one that would work with `_locale` properly?

Reply:
> I wasn't sure why it rewrites it to just locale (no underscore). Could you elaborate?

I think I was confused. I can't reproduce this anymore.

> Furthermore, could we overwrite the default `userLocaleMiddleware` middleware with our own one that would work with _locale properly?

You tell me! I don't know anything about `apiFetch` middleware other than that they exist. I've taken a quick look and I'm not sure what to suggest how to start. I think we also need the argument lest the resolver only be called the first time even if the `locale` changes (it's my understanding that the resolver will only be called the first time the selector is called with any given argument set?). Given that, I think it's a requirement that the locale be parameterized to the resolver.

Reply:
> I think I was confused. I can't reproduce this anymore.

👍  Yeah, can't reproduce it either, FWIW. 

> You tell me! I don't know anything about apiFetch middleware other than that they exist. I've taken a quick look and I'm not sure what to suggest how to start.

In general this would boil down to creating our own middleware just like [here](https://github.com/WordPress/gutenberg/blob/master/packages/api-fetch/src/middlewares/user-locale.js#L6) and doing a `apiFetch.use( ourUserLocaleMiddleware )` before executing the actual `apiFetch()`. 

Again, though, this is not a blocker, and definitely something we can try later if you're interested.